### DISCUSSION TITLE
🤔: React Nativeのデバッグ方法を知りたい

### DISCUSSION THREAD

**Discussion by takei0w0** [ws-4020/mobile-app-crib-notes#131] [Q&A]
学習コンテンツFBより

- モバイルアプリの開発でのデバッグはどのようにやるものなのかの紹介があるとうれしいかも、と思いました。私は主にiOSシュミレータで実装していたのでSafariを使いました。よく使われているのはどの方法なんだろう、と気になりました。
- デバッグ方法もしりたい（調べたのでわかったが色々ややこしかったので）
- 実際にやってみた感触として、デバッグどうするのかなという疑問がありました。何かデバッグのやり方でいい方法があれば、ぜひご共有いただきたいと思います。

**Comment by tani-takanori** [comment]
デバッグ方法の紹介はほしいけど、学習のステップに入れてしまうとその作業にとらわれてしまいそうなので紹介にとどめたい

**Reply by ghost** [reply]
Example Appのほうに「Development」枠を用意するので、そこに用意して演習課題でデバッグしてみてもらう、とかどうだろうか。

**Reply by takei0w0** [reply]
「Development」枠ってどういうイメージですかね？

**Reply by ghost** [reply]
ヘッダーから「React Native」＞「Example App」を開いたときのサイドバーに、「Development」ってのを用意しようと思ってます。

今のところ、「ビルドバリアント」と「依存パッケージの管理」ってページを入れる予定。そこに「アプリのデバッグ」みたいなページを用意したらいいんじゃないかなぁ、って思った。あと、今もMarkdownだけであれば「環境構築」ってやつがあるから、そこにデバッグ用の環境構築とかIDEのセットアップ方法とかを書いておいたら役に立ちそうかな、と。

現状のハンズオンで「VS Codeでデバッグ環境整えるところまでやります」はちと厳しそうかも？って思ったので。でも、自分で調べたりできるようになるには、デバッグのやり方って抑えておかないといけないかな？？

**Reply by takei0w0** [reply]
ハンズオンの環境構築に入れると混乱しそうかと思いましたが、たしかに自分で調べながらやりたいという人もいるかもしれませんね！

Example Appにページ用意しつつ、必要な人が参照できるように開発環境の構築からリンク貼るのがいいかなと思いました。

**Comment by takei0w0** [comment]
公式ページから以下のデバッグ方法ピックアップして紹介すれば開発には困らない気がする
https://reactnative.dev/docs/debugging

- Chromeでのデバッグ
- Safariでのデバッグ
- react-devtoolsを使ったデバッグ
  - react-devtoolsは使ったことないけどインスペクター見るなら使った方が便利そう

**Reply by ghost** [reply]
> Chromeでのデバッグ
> Safariでのデバッグ

個人的には、VS CodeやIntelliJ IDEAでのデバッグ方法（というか、設定方法）を紹介してあげるのがよさそうかな、って思った。あとは、[React Native Support | Flipper](https://fbflipper.com/docs/features/react-native/)かな。

**Reply by takei0w0** [reply]
たしかに 手元のエディターでできたほうがいいですね

Flipper初めて知ったので見てみます

**Comment by creature-water-valley** [comment]
以下に追加済み。
[Flipper](https://ws-4020.github.io/mobile-app-crib-notes/react-native/santoku/development/tools/flipper)

ただし、Flipperに変わるツールがリリースされそうではある。
https://reactnative.dev/blog/2023/12/06/0.73-debugging-improvements-stable-symlinks
